lib/publiccloud/instance.pm

```diff
@@ -432,9 +502,10 @@ sub softreboot {
     my $start_time = time();
 
     # wait till ssh disappear
-    while (($duration = time() - $start_time) < $args{timeout}) {
-        last unless (defined($self->wait_for_ssh(timeout => 1, proceed_on_failure => 1, username => $args{username})));
-    }
+    my $out = $self->wait_for_ssh(timeout => $args{timeout}, wait_stop => 1, username => $args{username});
+    # ok ssh port closed
+    record_info("STOP NOK", "$out") if (defined $out);    # not ok port still open
```

Comment:
?????????????

Reply:
In the case we check the ssh for **halting** systems,  we expect the $out is `undef`. The previous code didn't consider this check, and here I only provide a debug info in the case the system _still answering_ after timeout, that is $out is defined, instead.

Reply:
Or just update the `STOP NOK` with something that everybody can understand without reading the code.

Reply:
done